otwcode/otwarchive
# AO3-5872 Move RSpec tests back to the default directory structure

## Issue

https://otwarchive.atlassian.net/browse/AO3-5872

## Purpose

Move all specs back to the [directory structure inferred by RSpec](https://relishapp.com/rspec/rspec-rails/docs/directory-structure), and change Travis/Codeship to do [`rspec --exclude-pattern "spec/{controllers,models}/**/*.rb"`](https://relishapp.com/rspec/rspec-core/v/3-8/docs/configuration/exclude-pattern) instead of `rspec spec/miscellaneous`.

## References

`$TEST_GROUP` in .travis.yml [must be quoted](https://stackoverflow.com/questions/10067266/when-to-wrap-quotes-around-a-shell-variable) to avoid the shell interfering with `--exclude-pattern`'s value.

## Testing Instructions

Travis should pass. The new test group should output "849 examples" (844 previously for spec/miscellaneous + 5 TagTypeHelper specs).

Before merging, [Codeship "Test Pipelines"](https://github.com/otwcode/otwarchive/wiki/Codeship) should be updated from:
```sh
./script/codeship/try.sh rspec "bundle exec rspec spec/controllers"
./script/codeship/try.sh rspec "bundle exec rspec spec/miscellaneous"
./script/codeship/try.sh rspec "bundle exec rspec spec/models"
```
to:
```sh
./script/codeship/try.sh rspec "bundle exec rspec spec/controllers"
./script/codeship/try.sh rspec "bundle exec rspec spec/models"
./script/codeship/try.sh rspec "bundle exec rspec --exclude-pattern 'spec/{controllers,models}/**/*.rb'"
```
